src/coreclr/jit/lclmorph.cpp

```diff
@@ -1333,6 +1361,15 @@ class LocalAddressVisitor final : public GenTreeVisitor<LocalAddressVisitor>
                 return IndirTransform::LclFld;
             }
 
+#ifdef TARGET_64BIT
+            if (!isDef && varTypeIsSmall(indir) && varTypeIsIntegral(varDsc) && !varTypeIsLong(varDsc))
+#else // TARGET_64BIT
+            if (!isDef && varTypeIsSmall(indir) && varTypeIsIntegral(varDsc) && !varTypeIsLong(varDsc))
```

Comment:
#ifdef and #else have identical code.

Reply:
They will change. I needed to be sure only casts from INTs worked before I do LONGs on x64.